exchange/src/wrapper/main.cpp

```diff
@@ -97,6 +107,29 @@ catch_sigint(int)
     while (true) {}
 }
 
+std::optional<std::string>
+get_algorithm()
+{
+    std::string error_buffer;
+    std::string algorithm_string;
+    glz::json_t algorithm_message{};
+
+    std::getline(std::cin, algorithm_string);
+    auto error = glz::read_json(algorithm_message, algorithm_string);
+
+    if (error) {
+        std::string descriptive_error = glz::format_error(error, error_buffer);
+        log_e(wrapper_init, "glz::read_json() failed: {}", descriptive_error);
+        return {};
```

Comment:
throw a runtime error (which won't be caught, which is okay). We can't handle this case